### ISSUE TITLE
Test day!

### ISSUE THREAD

**Issue by Prototik** [Prototik/HoloEverywhere#201]
I announce test day!
In short - I ask you to test the new version before the release.
Will be just fine if we create 1.4 together with the fewest bugs, not so right?

Download the new version, read the guide on migrating to 1.4, report bugs - it's simple, realy.

Download new version:
`git clone https://github.com/ChristopheVersieux/HoloEverywhere.git -b dev`
Or download [archive](https://github.com/ChristopheVersieux/HoloEverywhere/archive/dev.zip)

[Migration on 1.4](https://github.com/ChristopheVersieux/HoloEverywhere/wiki/Migration-on-1.4)

Also, you can help the i10n of the demo application. Open a `demo/resources/strings.json` and translate those phrases that already have `ru` locale.

**Comment by ghost** [comment]
I have successfully migrated my app to version 1.4.

Do not forget to remove any mention of the old theme from the manifest!

**Comment by bkhall** [comment]
I have migrated two apps to 1.4 with no issues so far. Great work!

**Comment by httpdispatch** [comment]
Tested it a little bit more on 2.1 and 2.3.3 emulators. First tests give good results.

**Comment by soarcn** [comment]
i have migrated to 1.4, most of issues related to dialog. which at least working correct in 1.3

precondition: turn off "always use parent theme", unless the theme setting will be ignore.
- use "Theme.Dialog" for preferenceActitivity, the screen size is not correct. try to reproduce this, please use following code in demo app.
  setTitle("short"); //important
  addPreferencesFromResource(R.xml.preferences);
- background and frame will also be shown in dialogfragment even we set DialogFragment.STYLE_NO_FRAME.
- preference will lost after the upgrade
- I would suggest HE still use XML preference imp as default one, since json imp not support some feature.
- suggest turn off background music as default in demo app

And thanks for you great work to bring this lib to us!

**Comment by bkhall** [comment]
Attempting to update another app - getting ClassCastException on CheckedTextView when used in a layout file.

The layoutinflater chokes on it - looks like a bad dimension somewhere.

I using this layout file: android.R.layout.simple_list_item_multiple_choice

I copied the file into my project just to be sure, still fails with the stack trace below.

If I view the layout in the ADT with the HE theme applied, I get errors that indicate the attributes used in the widget are missing from the theme.

android:layout_height="?android:attr/listPreferredItemHeight"
android:gravity="center_vertical"
android:paddingLeft="?android:attr/listPreferredItemPaddingLeft"
android:paddingRight="?android:attr/listPreferredItemPaddingRight"
android:textAppearance="?android:attr/textAppearanceListItem" />

**Comment by Prototik** [comment]
@soarcn
 ![Screenshot](http://i.imgur.com/ZUQXc.png)  
Where?

**Comment by soarcn** [comment]
@Prototik  please test it in android 2.3, I can reproduce with my defy.
and Dialog activity defects can be fixed by add those items to Theme:
        <item name="windowMinWidthMajor">@dimen/dialog_min_width_major</item>
        <item name="windowMinWidthMinor">@dimen/dialog_min_width_minor</item>

till now haven't found the fix for FragmentDialog, I will upload screenshot later.

**Comment by Prototik** [comment]
@bkhall Update to latest version from dev and use R.layout.simple_list_item_***

**Comment by soarcn** [comment]
![](http://www.cocosw.com/1.png)
Dialog fragmentactivity in android 2.3

![](http://www.cocosw.com/2.png)
dialogfragment NO_FRAME

I use the test code from ABS fragment FragmentDialogSupport.java

**Comment by Prototik** [comment]
@soarcn i'm push fix for dialog theme, but i don't find original source code, when while processing windowMinWidthMajor/Minor, because i did so, i thought the right. Test please.

**Comment by bkhall** [comment]
@Prototik Getting closer...

The HE theme is missing these two attribute definitions:

Couldn't find theme resource listPreferredItemPaddingLeft for the current theme
Couldn't find theme resource listPreferredItemPaddingRight for the current theme
"?listPreferredItemPaddingLeft" in attribute "paddingLeft" is not a valid format.
"?listPreferredItemPaddingRight" in attribute "paddingRight" is not a valid format.

Add those attributes and then I think the simple_list_item_***\* layouts will work.

**Comment by Prototik** [comment]
@bkhall fixed.

**Comment by soarcn** [comment]
@Prototik 
the Activity dialog theme  seems working correct now. 

but the (fragment)dialog still not correct. as you can see in the second screenshot, with NO_FRAME theme, there should have no frame(backgroud picture) around the dialog.

I checked the code, and found the root cause in DialogFragment.java  168

```
public Dialog onCreateDialog(Bundle savedInstanceState) {
return new AlertDialog(getActivity(), getTheme());
}
```

the correct way is 

```
public Dialog onCreateDialog(Bundle savedInstanceState) {
return new Dialog(getActivity(), getTheme());
}
```

I compared the HE 1.3, seems this is a change in 1.4

I make a workaround in my app layer to override the onCreateDialog, in that way, working perfect.

I would suggest HE could use some testcase UI besides demo application, for example use apidemo application(View section), that would be a easy way to cover most of real case. even looks not that fancy.

**Comment by bkhall** [comment]
@Prototik thanks for the list item fix, looks beautiful.

new issue - this test case below is crashing:

@Override
public void onCreateContextMenu(ContextMenu menu, View v,
        ContextMenuInfo menuInfo) {
    getMenuInflater().inflate(R.menu.song_context_menu, menu);
}

@Override
public boolean onContextItemSelected(MenuItem item) {
    AdapterView.AdapterContextMenuInfo menuInfo = (AdapterContextMenuInfo) item.getMenuInfo(); // fails here
    Toast.makeText(getActivity(), String.valueOf(menuInfo.position), Toast.LENGTH_SHORT).show();

```
return true;
```

}

I'm getting a ClassCastException on the AdapterContextMenuInfo
11-17 08:43:04.569: E/AndroidRuntime(18932): java.lang.ClassCastException: android.widget.AdapterView$AdapterContextMenuInfo

I have these imports:
import org.holoeverywhere.widget.AdapterView;
import org.holoeverywhere.widget.AdapterView.AdapterContextMenuInfo;
import android.view.ContextMenu.ContextMenuInfo;
import com.actionbarsherlock.view.ContextMenu;
import com.actionbarsherlock.view.MenuItem;

The context menu displays fine, but when I select something in the menu, it crashes.

**Comment by Prototik** [comment]
@soarcn fixed via b5e25f4
@bkhall change imports on android.widget.AdapterView

**Comment by bkhall** [comment]
@Prototik Thank you, that worked... I just assumed that since it was available in HE, that it was the one I was supposed to use. I'll try all the options in the future before reporting an issue.
